api/channel.go

```diff
@@ -953,6 +954,37 @@ func getChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 
 }
 
+func getChannelByName(c *Context, w http.ResponseWriter, r *http.Request) {
+	params := mux.Vars(r)
+	channelname := params["channel_name"]
+
+	cchan := Srv.Store.Channel().GetByName(c.TeamId, channelname)
+
+	if cresult := <-cchan; cresult.Err != nil {
+		c.Err = cresult.Err
+		return
+	} else {
+		data := &model.Channel{}
```

Comment:
Actually why isn't this just a copy of `getChannel()` that returns a `model.ChannelData` which has a `model.Channel` and `model.ChannelMember` ?